Sum up to 3998 (MMMCMXCVIII)

JorgePatrick opened the pull request:
Calculator finished.
Implemented sum for roman numbers up to 3998
Implemented validations for roman numbers up to 3998

JorgePatrick committed:
Sum up to 3998 (MMMCMXCVIII)

Calculator finished.
Implemented sum for roman numbers up to 3998
Implemented validations for roman numbers up to 3998

chicofaccin commented:
👏 good job!

chicofaccin commented on src/test/java/com/jorgepatrick/RomanCalculatorIntegrationTest.java:
```diff
@@ -243,9 +243,3841 @@ private static Stream<Arguments> provideRomanNumbersForCalculation() {
                 Arguments.of("LXXXVII", "LXXXIX", "CLXXVI"),
                 Arguments.of("LXXXVIII", "LXXXIX", "CLXXVII"),
-                Arguments.of("LXXXIX", "LXXXIX", "CLXXVIII")
+                Arguments.of("LXXXIX", "LXXXIX", "CLXXVIII"),
```
Wow... Did you include all the possible numbers? That is not a good approach. You can reduce the number of parameters. You can find the patterns and remove the duplications

chicofaccin commented on src/test/java/com/jorgepatrick/NumberConverterTest.java:
```diff
@@ -208,9 +208,3842 @@ private static Stream<Arguments> provideRomanNumbersForConversion() {
                 Arguments.of(175, "CLXXV"),
                 Arguments.of(176, "CLXXVI"),
                 Arguments.of(177, "CLXXVII"),
-                Arguments.of(178, "CLXXVIII")
```
Same thing here. Do you really need to test all the numbers from 100 to 200? And from 200 to 300? Are not they the same from 1 to 100?

JorgePatrick committed:
Improvements

Variables used in only one method changed to locals
Some methods changed to static
Enum PowerOfTenNumbers deleted, since it was unnecessary
Tests optimized
